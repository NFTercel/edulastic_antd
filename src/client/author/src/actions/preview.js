import { PREVIEW_UPDATE_LIST, CHANGE_PREVIEW_TAB } from "../constants/actions";

export const updatePreviewListAction = list => ({
  type: PREVIEW_UPDATE_LIST,
  payload: { list }
});

export const changePreviewTabAction = previewTab => ({
  type: CHANGE_PREVIEW_TAB,
  payload: { previewTab }
});
