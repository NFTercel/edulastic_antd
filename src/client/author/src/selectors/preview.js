import { createSelector } from "reselect";

export const moduleName = "preview";

export const stateSelector = state => state[moduleName];
export const getPreviewIndexesListSelector = createSelector(
  stateSelector,
  state => state.list
);
export const getPreivewTabSelector = createSelector(
  stateSelector,
  state => state.previewTab
);
