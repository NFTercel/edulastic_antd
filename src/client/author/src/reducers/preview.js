import { PREVIEW_UPDATE_LIST, CHANGE_PREVIEW_TAB } from "../constants/actions";

const initialState = {
  list: [], // array of indexes,
  previewTab: "clear"
};

export default function reducer(state = initialState, { type, payload }) {
  switch (type) {
    case PREVIEW_UPDATE_LIST:
      return { ...state, list: payload.list };
    case CHANGE_PREVIEW_TAB:
      return { ...state, previewTab: payload.previewTab };

    default:
      return state;
  }
}
