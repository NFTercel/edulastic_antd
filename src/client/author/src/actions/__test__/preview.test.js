import { updatePreviewListAction, changePreviewTabAction } from "../preview";

describe("Preview", () => {
  it("update preview list should return an action", () => {
    expect(updatePreviewListAction({})).toMatchSnapshot();
  });

  it("change preview tab should return an action", () => {
    expect(changePreviewTabAction("check")).toMatchSnapshot();
    expect(changePreviewTabAction("show")).toMatchSnapshot();
    expect(changePreviewTabAction("clear")).toMatchSnapshot();
  });
});
