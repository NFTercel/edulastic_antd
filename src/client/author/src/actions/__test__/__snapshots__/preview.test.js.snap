// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview change preview tab should return an action 1`] = `
Object {
  "payload": Object {
    "previewTab": "check",
  },
  "type": "[preview] CHANGE_PREVIEW_TAB",
}
`;

exports[`Preview change preview tab should return an action 2`] = `
Object {
  "payload": Object {
    "previewTab": "show",
  },
  "type": "[preview] CHANGE_PREVIEW_TAB",
}
`;

exports[`Preview change preview tab should return an action 3`] = `
Object {
  "payload": Object {
    "previewTab": "clear",
  },
  "type": "[preview] CHANGE_PREVIEW_TAB",
}
`;

exports[`Preview update preview list should return an action 1`] = `
Object {
  "payload": Object {
    "list": Object {},
  },
  "type": "[preview] PREVIEW_UPDATE_LIST",
}
`;
